### PR TITLE
[RW-6558]: Add new endpoint in admin workspace for deleting runtime in this workspace

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:
Is is copied from https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/api/RuntimeController.java#L108
And https://github.com/all-of-us/workbench/blob/master/api/src/test/java/org/pmiops/workbench/api/RuntimeControllerTest.java#L718-L808

With some minor improvements

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/WorkspaceAdminController.java [hunk 1]
```diff
@@ -72,4 +74,12 @@ public WorkspaceAdminController(WorkspaceAdminService workspaceAdminService) {
   public ResponseEntity<List<FileDetail>> listFiles(String workspaceNamespace) {
     return ResponseEntity.ok(workspaceAdminService.listFiles(workspaceNamespace));
   }
+
+  @Override
+  @AuthorityRequired({Authority.RESEARCHER_DATA_VIEW})
```
Comment (yonghaoy) [inline api/src/main/java/org/pmiops/workbench/api/WorkspaceAdminController.java:79]: The original method is:
```
  @Override
  @AuthorityRequired(Authority.SECURITY_ADMIN)
```

Not sure if this is correct here.

Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/api/WorkspaceAdminController.java:79]: It should be security admin. The other methods here do not mutate anything, they are read-only operations, hence the different authority.

File: api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java [hunk 2]
```diff
@@ -293,4 +346,11 @@ private WorkspaceUserAdminView toWorkspaceUserAdminView(UserRole userRole) {
                 .role(userRole.getRole())
                 .userModel(userMapper.toApiUser(userRole, null)));
   }
+
+  private static Stream<LeonardoListRuntimeResponse> filterByRuntimesInList(
+      Stream<LeonardoListRuntimeResponse> clustersToFilter, List<String> runtimeNames) {
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java:351]: nit: probably an existing issue from my migration, but please rename cluster -> runtime here

File: api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java [hunk 3]
```diff
@@ -205,14 +217,55 @@ public WorkspaceAdminView getWorkspaceAdminView(String workspaceNamespace) {
         .resources(adminWorkspaceResources);
   }
 
+  @Override
+  public List<ListRuntimeResponse> deleteRuntimesInWorkspace(
+      String workspaceNamespace, ListRuntimeDeleteRequest req) {
+    final String googleProject =
+        getWorkspaceByNamespaceOrThrow(workspaceNamespace).getGoogleProject();
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java:224]: I'm assuming this is the only change to the implementation here

**Review by calbach — APPROVED** [review]

**Review by jmthibault79 — APPROVED** [review]

File: api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java (hunk 3 above)
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java:224]: yup. Extra db lookup to get googleProject

**Pull request merged**